soc/arm/microchip_mec/mec1501/soc.c

```diff
@@ -151,11 +168,40 @@ static int soc_init(struct device *dev)
 
 	soc_ecia_init();
 
+	soc_timer_init();
+
 	if (!isave) {
 		__enable_irq();
 	}
 
 	return 0;
 }
 
+#ifdef CONFIG_ARCH_HAS_CUSTOM_BUSY_WAIT
+
+#define DELAY_CALL_OVERHEAD_US 2
+
+/*
+ * 32-bit basic timer 0 has been configured for 16MHz auto-reload
+ * count up, no-interrupt mode.
+ */
+void z_arch_busy_wait(u32_t usec_to_wait)
+{
+	/* use 64-bit math to prevent overflow when multiplying */
+	u32_t cycles_to_wait = (u32_t)(
+		(u64_t)usec_to_wait << 4
```

Comment:
@scottwcpg  shouldn't this match consider the pre-scaler value used in initialization.
Changing this to 2, I get more precised delay for values between 20-50us